File: android/src/main/java/ru/innim/flutter_login_vk/Results.java
package ru.innim.flutter_login_vk;

import com.vk.api.sdk.auth.VKAccessToken;
import com.vk.sdk.api.base.dto.BaseBoolInt;
import com.vk.sdk.api.users.dto.UsersUserXtrCounters;

import java.util.HashMap;

public class Results {
    private enum LoginStatus {
        Success,
        Cancel,
        Error
    }

    public static HashMap<String, Object> loginCancelled() {
        return new HashMap<String, Object>() {{
            put("isCanceled", true);
        }};
    }

    public static HashMap<String, Object> loginSuccess(final VKAccessToken accessToken) {
        return new HashMap<String, Object>() {{
            put("accessToken", accessToken(accessToken));
        }};
    }

    public static HashMap<String, Object> error(final VKError error) {
        return new HashMap<String, Object>() {{
            put("apiCode", error.errorCode);
            put("message", error.errorMsg);
        }};
    }

    public static HashMap<String, Object> accessToken(final VKAccessToken accessToken) {
        if (accessToken == null)
            return null;
        return new HashMap<String, Object>() {{
            put("token", accessToken.getAccessToken());
            put("userId", String.valueOf(accessToken.getUserId()));
            put("created", accessToken.getCreated());
            put("email", accessToken.getEmail());
            put("isValid", accessToken.isValid());
            put("secret", accessToken.getSecret());
        }};
    }

    public static HashMap<String, Object> userProfile(final UsersUserXtrCounters user) {
        if (user == null)
            return null;

        return new HashMap<String, Object>() {{
            put("userId", user.getId().getValue());
            put("firstName", user.getFirstName());
            put("lastName", user.getLastName());
            put("online", user.getOnline() == BaseBoolInt.YES);
            put("onlineMobile", user.getOnlineMobile() == BaseBoolInt.YES);
            put("photo50", user.getPhoto50());
            put("photo100", user.getPhoto100());
            put("photo200", user.getPhoto200());
        }};
    }
}

File: android/src/main/kotlin/ru/innim/flutter_login_vk/Results.kt
package ru.innim.flutter_login_vk

import com.vk.api.sdk.auth.VKAccessToken
import com.vk.sdk.api.base.dto.BaseBoolInt
import com.vk.sdk.api.users.dto.UsersUserFull
import java.util.HashMap

object Results {
    fun loginCancelled(): HashMap<String, Any> {
        return hashMapOf("isCanceled" to true)
    }

    fun loginSuccess(accessToken: VKAccessToken): HashMap<String, Any> {
        return hashMapOf("accessToken" to accessToken(accessToken))
    }

    fun error(error: VKError): HashMap<String, Any?> {
        return hashMapOf(
                "apiCode" to error.errorCode,
                "message" to error.errorMsg
        )
    }

    fun accessToken(accessToken: VKAccessToken): HashMap<String, Any?> {
        return hashMapOf(
                "token" to accessToken.accessToken,
                "userId" to accessToken.userId.toString(),
                "created" to accessToken.created,
                "email" to accessToken.email,
                "isValid" to accessToken.isValid,
                "secret" to accessToken.secret
        )
    }

    fun userProfile(user: UsersUserFull): HashMap<String, Any?> {
        return hashMapOf(
                "userId" to user.id.value,
                "firstName" to user.firstName,
                "lastName" to user.lastName,
                "online" to (user.online == BaseBoolInt.YES),
                "onlineMobile" to (user.onlineMobile == BaseBoolInt.YES),
                "photo50" to user.photo50,
                "photo100" to user.photo100,
                "photo200" to user.photo200
        )
    }
}
